config/v1/types_infrastructure.go

```diff
@@ -123,6 +123,20 @@ const (
 	ExternalTopologyMode TopologyMode = "External"
 )
 
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+// +kubebuilder:validation:Enum:="";"Internal";"External"
+type PlatformLoadBalancerType string
+
+const (
+	// LoadBalancerTypeExternal is a load balancer that is managed outside of the cluster.
+	LoadBalancerTypeExternal PlatformLoadBalancerType = "External"
+	// LoadBalancerTypeInternal is a load balancer that is managed by the cluster.
+	LoadBalancerTypeInternal PlatformLoadBalancerType = "Internal"
+	// LoadBalancerTypeDefault indicates the default load balancer type for the platform.
+	// We let the cluster to choose the default load balancer type.
+	LoadBalancerTypeDefault PlatformLoadBalancerType = ""
```

Comment:
I'm not sure if I need something like `LoadBalancerTypeDefault` since we omit the field if empty.